### ISSUE TITLE
geordi dump should stop on errors

### ISSUE THREAD

**Issue by tessi** [makandra/geordi#32]
I executed `geordi dump production` recently where my postgres was set up in a wrong way (the database cluster was already on 9.6, where my pg binaries were still on version 9.5) which lead to an error while producing the dump:

```
geordi dump production

# Dumping the database of production
> Connecting to my.made-up-domain.de
> Dumping database for "production" environment ...
pg_dump: server version: 9.6.2; pg_dump version: 9.5.6
pg_dump: aborting because of server version mismatch
> Dumped to /home/made_up_user/dumps/dump_for_download.dump (0 KB)
Connection to my.made-up-domain.de closed.
> Downloading remote dump to tmp/production.dump ...
dump_for_download.dump                                                                                                                          100%    0     0.0KB/s   00:00

> Dumped the production database to tmp/production.dump.
```

Geordi continued execution after `pg_dump` failed with an error and downloaded an empty file to `tmp/production.dump`.
Instead, I would expect geordi to abort once the `pg_dump` command failed and immediately return with an error code.

**Comment by codener** [comment]
You are right, I noticed that issue some time ago. I'll be happy to discuss a pull request!

**Comment by tessi** [comment]
I made an attempt to fix this at #33
